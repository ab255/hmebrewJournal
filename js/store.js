const store = [
  {beerName: 'Green Bomber', beerType: 'Imperial IPA', brewDate: '4/28/2013'},
  {beerName: 'Red Dawn', beerType: 'American Amber Ale', brewDate: '3/26/2013'}
];

module.exports = store;
