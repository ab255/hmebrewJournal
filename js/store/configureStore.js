import { AsyncStorage } from 'react-native';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, autoRehydrate } from 'redux-persist';

import reducers from '../reducers/reducers';


const createBrewStore = applyMiddleware(thunk)(createStore);

const configureStore = () => {
  const store = autoRehydrate()(createBrewStore)(reducers);
  persistStore(store, {storage: AsyncStorage});
  return store;
}

export default configureStore;
