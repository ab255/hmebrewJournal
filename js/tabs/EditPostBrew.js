import React, {Component} from 'react';
import {
  DatePickerIOS,
  ScrollView,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import moment from 'moment';

class EditPostBrew extends Component {
  constructor(props) {
    super(props);
    this.state = {
      DateStartFermentation: this.props.dateStartFermentation || new Date(),
      FermentationNotes: this.props.fermentationNotes,
      PackagingDate: this.props.packagingDate || new Date(),
      HydrometerReading: this.props.hydrometerReading,
      TypeOfPackaging: this.props.typeOfPackaging,
      NotesAboutPackaging: this.props.notesAboutPackaging,
      datePickerModeOne:'hidden',
      datePickerModeTwo: 'hidden',
    }
  }

  toggleDatePickerOne = () => {
    let mode = this.state.datePickerModeOne =='hidden' ? 'visible' : 'hidden';
    this.setState({ datePickerModeOne: mode });
  }

  onDateChangeOne = (dateOne) => {
    this.setState({ DateStartFermentation: dateOne })
  }

  toggleDatePickerTwo = () => {
    let mode = this.state.datePickerModeTwo =='hidden' ? 'visible' : 'hidden';
    this.setState({ datePickerModeTwo: mode });
  }

  onDateChangeTwo = (dateTwo) => {
    this.setState({ PackagingDate: dateTwo })
  }

  render() {
    let datePickerOne = (
      <View style={styles.datePicker}>
        <DatePickerIOS
          date={new Date(this.state.DateStartFermentation)}
          mode='date'
          onDateChange={this.onDateChangeOne.bind(this)}
          stlye={styles.datePicker}
        />
      </View>
    )
    let datePickerTwo = (
      <View style={styles.datePicker}>
        <DatePickerIOS
          date={new Date(this.state.PackagingDate)}
          mode='date'
          onDateChange={this.onDateChangeTwo.bind(this)}
          stlye={styles.datePicker}
        />
      </View>
    )
    return (
      <KeyboardAwareScrollView style={styles.container}>
        <View>
          <View>
            <Text style={styles.title}>Fermentation Start:</Text>
              <TouchableWithoutFeedback onPress={this.toggleDatePickerOne.bind(this) }>
                <View style={styles.dateField}>
                  <Text style={styles.dateText}>
                    {moment(this.state.DateStartFermentation).format('MMMM Do YYYY')}
                  </Text>
                </View>
              </TouchableWithoutFeedback>
            </View>
            { this.state.datePickerModeOne == 'visible' ? datePickerOne : <View/> }
            <Text style={styles.title}>(Click Date To Toggle Date Picker)</Text>
        </View>
        <View>
          <Text style={styles.title}>Fermentation Notes:</Text>
          <TextInput
            multiline={true}
            style={styles.ingredientsInput}
            onChangeText={(FermentationNotes) => this.setState({FermentationNotes})}
            value={this.state.FermentationNotes}
            placeholder='Fermentation notes'
          />
        </View>
        <View>
          <View>
            <Text style={styles.title}>Packaging Date:</Text>
              <TouchableWithoutFeedback onPress={this.toggleDatePickerTwo.bind(this) }>
                <View style={styles.dateField}>
                  <Text style={styles.dateText}>
                    {moment(this.state.PackagingDate).format('MMMM Do YYYY')}
                  </Text>
                </View>
              </TouchableWithoutFeedback>
            </View>
            { this.state.datePickerModeTwo == 'visible' ? datePickerTwo : <View/> }
            <Text style={styles.title}>(Click Date To Toggle Date Picker)</Text>
        </View>
        <View>
          <Text style={styles.title}>Final Gravity:</Text>
          <TextInput
            style={styles.brewInput}
            onChangeText={(HydrometerReading) => this.setState({HydrometerReading})}
            value={this.state.HydrometerReading}
            placeholder='Final gravity ex. 1.026'
            keyboardType='decimal-pad'
          />
        </View>
        <View>
          <Text style={styles.title}>Type of Packaging:</Text>
          <TextInput
            style={styles.brewInput}
            onChangeText={(TypeOfPackaging) => this.setState({TypeOfPackaging})}
            value={this.state.TypeOfPackaging}
            placeholder='Type of packaging'
          />
        </View>
        <View>
          <Text style={styles.title}>Packaging Notes:</Text>
          <TextInput
            multiline={true}
            style={styles.ingredientsInput}
            onChangeText={(NotesAboutPackaging) => this.setState({NotesAboutPackaging})}
            value={this.state.NotesAboutPackaging}
            placeholder='Notes about packaging'
          />
        </View>
        <TouchableHighlight
        >
          <View style={styles.submitButton}>
            <Text style={styles.buttonText}>Submit</Text>
          </View>
        </TouchableHighlight>
      </KeyboardAwareScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#911F27',
    flex: 1,
    padding: 10,
  },
  dateField: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    height: 30,
    justifyContent: 'center',
    padding: 5,
    marginBottom: 5,
  },
  dateText: {
    color: '#630A10',
    fontSize: 16,
  },
  datePicker: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    marginBottom: 5,
  },
  title: {
    color: '#F7D098',
    fontWeight: 'bold',
    marginBottom: 5,
  },
  brewInput: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    color: '#630A10',
    height: 30,
    paddingLeft: 5,
    marginBottom: 10,
  },
  ingredientsInput: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    color: '#630A10',
    paddingLeft: 5,
    height: 100,
    fontSize: 16,
    marginBottom: 10,
  },
  submitButton: {
    alignSelf: 'center',
    backgroundColor: '#F7D098',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 5,
    width: 300,
    height: 30,
    margin: 20,
    },
  buttonText: {
    color: '#630A10',
    fontSize: 16,
  },
})

export default EditPostBrew;
