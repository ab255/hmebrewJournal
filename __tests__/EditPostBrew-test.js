import 'react-native';
import React from 'react';
import EditPostBrew from '../js/tabs/EditPostBrew.js';
import store from '../js/mockStore.js'

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  let beer = store
  const tree = renderer.create(
    <EditPostBrew route={ { beer } } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
