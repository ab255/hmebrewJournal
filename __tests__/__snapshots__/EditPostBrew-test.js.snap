exports[`test renders correctly 1`] = `
<ScrollView
  contentInset={
    Object {
      "bottom": 49,
    }
  }
  enableAutoAutomaticScroll={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  onScroll={[Function]}
  scrollEventThrottle={8}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "#911F27",
      "flex": 1,
      "padding": 10,
    }
  }>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#F7D098",
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }>
        Fermentation Start:
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderRadius": 5,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "padding": 5,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#630A10",
              "fontSize": 16,
            }
          }>
          May 7th 2013
        </Text>
      </View>
    </View>
    <View />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      (Click Date To Toggle Date Picker)
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Fermentation Notes:
    </Text>
    <TextInput
      multiline={true}
      onChangeText={[Function]}
      placeholder="Fermentation notes"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "fontSize": 16,
          "height": 100,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="" />
  </View>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#F7D098",
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }>
        Packaging Date:
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderRadius": 5,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "padding": 5,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#630A10",
              "fontSize": 16,
            }
          }>
          November 10th 2016
        </Text>
      </View>
    </View>
    <View />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      (Click Date To Toggle Date Picker)
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Final Gravity:
    </Text>
    <TextInput
      keyboardType="decimal-pad"
      onChangeText={[Function]}
      placeholder="Final gravity ex. 1.026"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "height": 30,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="1.015" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Type of Packaging:
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Type of packaging"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "height": 30,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Packaging Notes:
    </Text>
    <TextInput
      multiline={true}
      onChangeText={[Function]}
      placeholder="Notes about packaging"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "fontSize": 16,
          "height": 100,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="" />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F7D098",
          "borderRadius": 5,
          "height": 30,
          "justifyContent": "center",
          "margin": 20,
          "width": 300,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#630A10",
            "fontSize": 16,
          }
        }>
        Submit
      </Text>
    </View>
  </View>
</ScrollView>
`;
